source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -119,7 +119,7 @@ void AuthenticatorImpl::verify(Http::HeaderMap& headers, Authenticator::Callback
   const auto unix_timestamp = std::chrono::duration_cast<std::chrono::seconds>(
                                   std::chrono::system_clock::now().time_since_epoch())
                                   .count();
-  if (jwt_.exp_ < unix_timestamp) {
+  if (jwt_.exp_ > 0 && jwt_.exp_ < unix_timestamp) {
```

Comment:
@qiwzhang does jwt_verify_lib assign 0 to `exp_` when "exp" is not in the JWT?

Even if so, "0" is valid "exp", perhaps we can make `exp_` an `absl::optional<uint64_t>`?

Reply:
@ccaraman Fixed DCO and added a test thanks!
@lizan jwt_verify_lib assigns 0 to exp_ if not specified (Reference: https://github.com/google/jwt_verify_lib/blob/b06ba30f5d2049bcabd6b4ae8bb8f3518fb18b44/src/jwt.cc)

Reply:
OK thanks, can you add a comment to clarify the 0 case?